source/Ship.cpp

```diff
@@ -2669,6 +2669,34 @@ bool Ship::IsLanding() const
 
 
 
+float Ship::LandingSpeed() const
+{
+	// Makes no sense to query the landing speed if we are not landing.
+	// Return the default value instead.
+	if(!IsLanding())
+		return 0.02f;
+
+	const float mass = Mass();
+	const float landingSpeed = attributes.Get("landing speed");
+
+	const StellarObject *planet = landingPlanet->GetSystem()->FindStellar(landingPlanet);
+	const auto planetAttributes = landingPlanet->Attributes();
+	const float planetLandingSpeed = planetAttributes.count("very fast landing") ? 1.4f :
+		planetAttributes.count("fast landing") ? 1.2f :
+		(planet->IsMoon() || planetAttributes.count("slow landing")) ? .8f :
+		(planet->IsStation() || planetAttributes.count("very slow landing")) ? .6f :
```

Comment:
```suggestion
	const float planetLandingSpeed = planetAttributes.count("very slow landing") ? .8f :
		planetAttributes.count("slow landing") ? .9f :
		(planet->IsMoon() || planetAttributes.count("fast landing")) ? 1.1f :
		(planet->IsStation() || planetAttributes.count("very fast landing")) ? 1.2f :
```

Reply:
I really don't have any opinion on how this should go, other than how it looks.
This indeed makes the most sense but will it look good?

Reply:
Also what should we do for gas giants, make them slower by default?
What about ring worlds and wormholes? Currently they use the default too.